src/resolve/metadataResolver.ts

```diff
@@ -369,8 +369,21 @@ const resolveType =
  */
 const parseAsContentMetadataXml =
   (registry: RegistryAccess) =>
-  (fsPath: string): boolean =>
-    Boolean(registry.getTypeBySuffix(extName(fsPath)));
+  (fsPath: string): boolean => {
+    const suffixType = registry.getTypeBySuffix(extName(fsPath));
+    if (!suffixType) return false;
+
+    const matchesSuffixType = fsPath.split(sep).includes(suffixType.directoryName);
+    if (matchesSuffixType) return matchesSuffixType;
+
+    // it might be a type that requires strict parent folder name.
+    const strictFolderSuffixType = registry
+      .getStrictFolderTypes()
+      .find((l) => l.suffix === suffixType.suffix && l.directoryName && l.name !== suffixType.name);
```

Comment:
if `registry.getTypeBySuffix` entry didn't match, check for strictFolderTypes.

snapshot tests caught an issue with `CustomSite` where `getTypeBySuffix` would return `sitedotcom` (no match with the `directoryName`), this will return the strict folder types -> find if one with directoryName and suffix matches (and isn't `suffixType`).
https://github.com/forcedotcom/source-deploy-retrieve/blob/9d94840fa9523c2e9d4268f231c7f889ab8160b0/src/registry/metadataRegistry.json#L459